src/targets/gpu/kernels/include/migraphx/kernels/reduce.hpp

```diff
@@ -105,7 +120,23 @@ __device__ auto block_reduce(index idx, Op op, T init, Index n, F f)
 #endif
     using type = decltype(index::invoke_loop(f, 0, _c<0>));
     __shared__ type buffer[idx.max_nlocal() / lanes_per_thread];
+    // Hack: Note when in int32_t and lowest/highest op, I need to overrid the init value
     type x = init;
+    type y = init;
+    if constexpr(is_same<type, int32_t>{})
+    {
+        if constexpr(is_same<T, lowest>{})
+        {
+            x = INT_MIN;
+            y = INT_MIN;
+        }
+        else if constexpr(is_same<T, highest>{})
+        {
+            x = INT_MAX;
+            y = INT_MAX;
```

Comment:
And this could be a bug in `numeric_max`. Is that actually needed? Or was this added for completeness?

Reply:
FYI, printing out the x as-is in %d (without the hack) will give me a mysterious `-260046824`.